File: 2372-rearrange-characters-to-make-target-string/rearrange-characters-to-make-target-string.c
int rearrangeCharacters(char* s, char* target) {
    int num_copies = 0;
    bool found;
    while (true) {
        for (int i = 0; target[i] != '\0'; i++) {
            found = false;
            for (int j = 0; s[j] != '\0'; j++) {
                if (target[i] == s[j]) {
                    found = true;
                    s[j] = '.';
                    break;
                }
            }
            if (!found)
                return num_copies;
        }
        num_copies++;
    }
    return num_copies;
}

File: 2372-rearrange-characters-to-make-target-string/rearrange-characters-to-make-target-string.py
class Solution:
    def rearrangeCharacters(self, s: str, target: str) -> int:
        s = list(s)
        num_copies = 0
        while True:
            for c in target:
                if c not in s:
                    return num_copies
                s.pop(s.index(c))
            num_copies += 1
        return num_copies
